ArcTMDb/ArcTMDb/ArcTMDb/ViewModels/SearchMoviesViewModel.cs

```diff
@@ -0,0 +1,119 @@
+﻿using ArcTMDb.Models;
+using ArcTMDb.Service;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Xamarin.Forms;
+
+namespace ArcTMDb.ViewModels
+{
+    public class SearchMoviesViewModel : BaseViewModel
+    {
+        private readonly IArcTMDbApiService _arcTMDbApiService;
+       
+        private GenreResults _genresResults;
+
+        private int page = 0;
+
+        private string _searchTerm;
+        public string SearchTerm
+        {
+            get { return _searchTerm; }
+            set
+            {
+                if (SetProperty(ref _searchTerm, value))
+                    SearchCommand.ChangeCanExecute();
+            }
+        }
+
+        public ObservableCollection<MovieDetails> SearchMovies { get; set; }
+
+        public Command SearchCommand { get; }
+
+        public Command<MovieDetails> LoadMoreMoviesCommand { get; set; }
+
+        public Command<MovieDetails> ShowMovieDetailsCommand { get; }
+
+        public SearchMoviesViewModel(IArcTMDbApiService arcTMDbApiService)
+        {
+            _arcTMDbApiService = arcTMDbApiService;
+
+            GetGenres();
+
+            SearchMovies = new ObservableCollection<MovieDetails>();
+
+            SearchCommand = new Command(ExecuteSearch, CanExecuteSearch);
+
+            LoadMoreMoviesCommand = new Command<MovieDetails>(ExecuteLoadMoreMovies, CanExecuteLoadMoreMovies);
+
+            ShowMovieDetailsCommand = new Command<MovieDetails>(ExecuteShowMovieDetails);
+        }
+
+        void ExecuteSearch()
+        {
+            page = 0;
+
+            SearchMovies.Clear();
+
+            LoadSearchMovies();
+        }
+
+        bool CanExecuteSearch()
+        {
+            return string.IsNullOrWhiteSpace(SearchTerm) == false;
+        }
+
+        private void ExecuteLoadMoreMovies(MovieDetails movieDetails)
+        {
+            LoadSearchMovies();
+        }
+
+        private async void LoadSearchMovies()
```

Comment:
I did the Xamarin marathon course and the Microsft instructor wrote some methods in the viewmodel using async void instead of async Task and he didn't use the suffixed async. So, I thought that was a good practice.